app/ts/library/provider.ts

```diff
@@ -71,28 +73,38 @@ export const connectBrowserProvider = async (
 		if (accounts.length === 0) {
 			removeProvider(store)
 		} else {
-			store.value = store.value ? { ...store.value, walletAddress: utils.getAddress(accounts[0]) } : undefined
+			const parsedAddress = AddressParser.parse(getAddress(accounts[0]))
+			if (!parsedAddress.success) throw new Error("Provider provided invalid address!")
+			store.value = store.value ? { ...store.value, walletAddress: parsedAddress.value } : undefined
 		}
 	}
 	const chainChangedCallback = async (chainId: string) => {
-		if ([1, 5].includes(Number(chainId))) {
+		if ([1n, 5n].includes(BigInt(chainId))) {
 			batch(() => {
-				appSettings.value = { ...appSettings.peek(), relayEndpoint: Number(chainId) === 1 ? MEV_RELAY_MAINNET : MEV_RELAY_GOERLI }
-				store.value = store.value ? { ...store.value, chainId: Number(chainId) } : undefined
+				appSettings.value = { ...appSettings.peek(), relayEndpoint: BigInt(chainId) === 1n ? MEV_RELAY_MAINNET : MEV_RELAY_GOERLI }
+				store.value = store.value ? { ...store.value, chainId: BigInt(chainId) } : undefined
 			})
 		} else {
-			store.value = store.value ? { ...store.value, chainId: Number(chainId) } : undefined
+			store.value = store.value ? { ...store.value, chainId: BigInt(chainId) } : undefined
+		}
+
+		const [accounts, block] = await Promise.all([provider.listAccounts(), provider.getBlock('latest')])
+		if (accounts.length > 0 && window.ethereum) {
+			clearEvents()
+			window.ethereum.on('disconnect', disconnectEventCallback)
+			window.ethereum.on('accountsChanged', accountsChangedCallback)
+			window.ethereum.on('chainChanged', chainChangedCallback)
+			provider.on('block', blockCallback)
 		}
```

Comment:
This feels unrelated to the move to Ethers v6.  If that is the case, can it be extracted into a separate PR so it can be evaluated separately?

Reply:
Had to change some of the types and methods when updating to v6, I also grabbed this bit from the provider.ts in nft-wallet is already in ethers v6. There was a bug when network changes the other events stop working, so assuming the the user is still connected we just clear and re-create the event listeners.